pkg/controllers/reallocation/controller.go

```diff
@@ -24,8 +24,8 @@ import (
 	"golang.org/x/time/rate"
 	"knative.dev/pkg/logging"
 
+	"github.com/benbjohnson/clock"
```

Comment:
You absolutely don't want a fake clock in our real code. Make sure this is only included in the test imports. This will ensure that golang's treeshaking doesn't build this package into our binary.